### PR TITLE
block builder internal message signing

### PR DESCRIPTION
**Pull request by frrist**

### CODE REVIEW

File: drivers/tipset_message_producer.go [hunk 1]
```diff
@@ -141,23 +133,55 @@ func (bb *BlockBuilder) WithBLSMessageAndRet(bm *types.Message, retval []byte) *
...
+
 func (bb *BlockBuilder) WithTicketCount(count int64) *BlockBuilder {
 	bb.ticketCount = count
 	return bb
 }
 
+func (bb *BlockBuilder) toSignedMessage(m *types.Message) *types.SignedMessage {
+	from := m.From
+	if from.Protocol() != address.SECP256K1 {
+		bb.TD.T.Fatalf("message From field must be SECP256K1 protocol, was: %v", from.Protocol())
```
Comment (anorth) [inline drivers/tipset_message_producer.go]: Sorry, this restriction prevents us testing important valid cases when `From` is an ID address of an account actor. We need to be able to resolve the ID address to corresponding account.

File: drivers/tipset_message_producer.go [hunk 2]
```diff
@@ -90,8 +92,9 @@ type Result struct {
 	ReturnVal []byte
 }
 
-func NewBlockBuilder(miner address.Address) *BlockBuilder {
+func NewBlockBuilder(miner address.Address, td *TestDriver) *BlockBuilder {
```
Comment (anorth) [inline drivers/tipset_message_producer.go]: Swap parameter order

**Review by anorth — CHANGES_REQUESTED** [review]

File: drivers/tipset_message_producer.go (hunk 1 above)
Comment (frrist) [reply drivers/tipset_message_producer.go]: Would you expect that lookup be performed by inspecting the init actors state or should chain-validation keep its own mapping?

**Comment by anorth** [comment]
Probably the latter

File: drivers/tipset_message_producer.go (hunk 1 above)
Comment (frrist) [reply drivers/tipset_message_producer.go]: Addressed in latest set of changes.

File: drivers/state.go [hunk 3]
```diff
@@ -32,11 +32,14 @@ type StateDriver struct {
 	st state.VMWrapper
 	w  state.KeyManager
 	rs state.RandomnessSource
+
+	// Mapping for IDAddresses to their pubkey/actor addresses. Used for lookup when signing messages.
+	ActorIDMap map[address.Address]address.Address
```
Comment (anorth) [inline drivers/state.go]: Don't export this. Provide a function if lookup from outside is needed.

**Review by anorth — APPROVED** [review]

**Pull request merged**